maud/data_model/maud_config.py

```diff
@@ -54,10 +57,22 @@ class MaudConfig:
     user_inits_file: Optional[str] = None
     ode_config: ODEConfig = Field(default_factory=ODEConfig)
     reject_non_steady: bool = True
+    penalize_non_steady: bool = False
     steady_state_threshold_abs: float = 1e-8
     steady_state_threshold_rel: float = 1e-3
+    steady_state_threshold_opt: Optional[dict[str, float]] = None
```

Comment:
As mentioned below I think this would be better if called `steady_state_penalty_sigma` or similar. 

I also think it's a little messy to have this be a dictionary and default to the absolute steady state threshold at the point when the stan input dictionary is created. If it's definitely necessary to have it vary per mic, could we perhaps use the same penalty but rescale based on the concentration as in the [relative check](https://github.com/biosustain/Maud/blob/dcd456473ba45ed9ff3ff4a29f6aaaa09b182980/maud/stan/functions.stan#L68)?

Reply:
I can get behind this. I guess it makes sense when you have multiple conditions where the concentration varies over several orders of magnitude. 

My only question would be how would the optimise behave given a varying probability function? I could imagine it would make it difficult to find the optimal given that the optimal changes. However, given that this is not inference it might provide a better initial approach to the steady state region.